tests/10-deploy

```diff
@@ -7,30 +7,61 @@ import unittest
 class TestDeployment(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        cls.deployment = amulet.Deployment(series='xenial')
-        cls.deployment.add('etcd')
-        try:
-            cls.deployment.setup(timeout=900)
-            cls.deployment.sentry.wait()
-        except amulet.helpers.TimeoutError:
-            msg = "Environment wasn't stood up in time"
-            amulet.raise_status(amulet.SKIP, msg=msg)
-        except:
-            raise
-
-    def test_single_service(self):
-        status = self.deployment.sentry['etcd'][0].run('service etcd status')
+        cls.d = amulet.Deployment(series='xenial')
+        cls.d.add('etcd')
+        cls.d.setup(timeout=900)
+        cls.d.sentry.wait()
+        cls.etcd = cls.d.sentry['etcd']
+        # find the leader
+        for unit in cls.etcd:
+            leader_result = unit.run('is-leader')
+            if leader_result[0] == 'True':
+                cls.leader = unit
+
+    def test_leader_status(self):
+        ''' Verify our leader is running the etcd daemon '''
+        status = self.leader.run('service etcd status')
         self.assertTrue("running" in status[0])
 
-    def test_two_node_scale(self):
-        self.deployment.add_unit('etcd')
-        self.deployment.sentry.wait()
+    def test_node_scale(self):
+        ''' Scale beyond 1 node because etcd supports peering as a standalone
+        application.'''
+        # Ensure we aren't testing a single node
+        if not len(self.etcd) > 1:
+            self.d.add_unit('etcd')
+            self.d.sentry.wait()
+
+        for unit in self.etcd:
+            status = unit.run('service etcd status')
+            self.assertFalse(status[1] == 1)
+            self.assertTrue("running" in status[0])
+
+    def test_cluster_health(self):
+        ''' Iterate all the units and verify we have a clean bill of health
+        from etcd '''
+        for unit in self.etcd:
+            health = unit.run('etcdctl cluster-health')
```

Comment:
we should call health.lower() here. Otherwise "Unhealthy" will not mach "unhealthy" right?